packages/cache/src/cache.ts

```diff
@@ -304,8 +317,8 @@ async function restoreCacheV2(
     if (typedError.name === ValidationError.name) {
       throw error
     } else {
-      // Supress all non-validation cache related errors because caching should be optional
-      core.warning(`Failed to restore: ${(error as Error).message}`)
+      // Log cache related errors
+      core.error(`Failed to restore: ${(error as Error).message}`)
```

Comment:
Add a specific condition for server errors (5xx) that will emit an error log and keep all the other errors as warnings. Fix all the unit tests to match this behaviour.